src/jira/client/axios.ts

```diff
@@ -149,13 +149,13 @@ const instrumentRequest = (response) => {
  * @param logger
  */
 const instrumentFailedRequest = (instance: AxiosInstance, logger: Logger) => {
-	return async (error) => {
+	return async (error: AxiosError) => {
 		instrumentRequest(error?.response);
-		if (error?.statusCode === 503) {
+		if (error.response?.status === 503) {
 			const statusResponse: AxiosResponse = await instance.get("/status");
 			if (statusResponse?.status === 503) {
 				logger.info(`503 from Jira: Jira instance ${instance} is suspended or does not exist. Returning 404 to our application.`);
-				error.statusCode = 404;
+				error.response.status = 404;
```

Comment:
Looks like this codepath is not cover by any tests. Do you think it worth adding one?

Reply:
yeah, I think there's some value in adding a test for this codepath as it would have caught the issue before going to production.